pallets/parachain-staking/src/lib.rs

```diff
@@ -1310,7 +1327,7 @@ pub mod pallet {
 			);
 
 			// we don't unlock immediately
-			Self::prep_unstake(&collator, less)?;
+			Self::prep_unstake(&collator, less, T::MaxUnstakeRequests::get().saturating_sub(1))?;
```

Comment:
a comment for why we do the sub 1 might be good. this is not immediately obvious i think.